netshare/utils/field.py

```diff
@@ -148,14 +176,28 @@ def denormalize(self, bin_x):
 
         return decimal_x
 
+    def denormalize(self, bin_x):
+        return np.array([self._denormalize(b) for b in bin_x])
+
     def getOutputType(self):
         outputs = []
 
         for i in range(self.num_bits):
-            outputs.append(
-                Output(
-                    type_=OutputType.DISCRETE,
-                    dim=2
-                ))
+            outputs.append(Output(type_=OutputType.DISCRETE, dim=2))
 
         return outputs
+
+    def getOutputDim(self) -> int:
+        return 2 * self.num_bits  # type: ignore
+
+
+def field_config_to_key(field: Config) -> FieldKey:
+    return (
```

Comment:
ohh because this object is not aimed to "hold" data, but only as a key to a hashmap.
I don't think that people should use the content of the object that is returned here 🤔

Reply:
Got it. That makes sense! Thanks for the explanations!